java-frontend/src/main/java/org/sonar/java/resolve/ParametrizedTypeJavaType.java

```diff
@@ -30,7 +30,7 @@
 
 public class ParametrizedTypeJavaType extends ClassJavaType {
 
-  private static TypeSubstitutionSolver typeSubstitutionSolver;
+  public static TypeSubstitutionSolver typeSubstitutionSolver;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Make this "public static typeSubstitutionSolver" field final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1444)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Make typeSubstitutionSolver a static final constant or non-public and provide accessors if needed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)